### PR TITLE
Transition to IC.compile

### PR DESCRIPTION
**Pull request by wpopielarski**
currently EclipseSbtBuildManager uses deprecated AggressiveCompile
class. This fix utilizes IC.incrementalCompile to be a replacement
for deprecated class.

### CODE REVIEW

**Comment by sschaef** [comment]
retest this please

File: org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/CachingCompiler.scala [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import sbt.compiler.AnalyzingCompiler
+import sbt.compiler.IC
+import sbt.compiler.MixedAnalyzingCompiler
+import sbt.inc.Analysis
+import sbt.inc.AnalysisStore
+import xsbti.Logger
+import xsbti.Reporter
+import xsbti.compile.Compilers
+
+package org.scalaide.core.internal.builder.zinc {
```
Comment (sschaef) [inline org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/CachingCompiler.scala]: Please add the package declaration at the top and move the EmptyAnalysis file to its own file. It is annoying if one needs to find classes, which can't easily be found through the file system.

File: org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/CachingCompiler.scala [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      } getOrElse {
+        (EmptyAnalysis(initialNameHashing), None)
+      }
+    }
+
+    private[zinc] def apply(cacheFile: File): AnalysisStore =
+      MixedAnalyzingCompiler.staticCachedStore(cacheFile)
+  }
+
+  class CachingCompiler private (val cacheFile: File, val sbtReporter: Reporter, val log: Logger) {
```
Comment (sschaef) [inline org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/CachingCompiler.scala]: It doesn't look like the ctor params need to be fields.

File: org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/EmptyAnalysis.scala [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import sbt.inc.Analysis
+
+package sbt {
+  /** Just to expose `Analysis.empty(nameHashing: Boolean)` to non `sbt` packages. */
```
Comment (Kwestor) [inline org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/EmptyAnalysis.scala]: Can you add a ticket to sbt to expose this and add ticket number here?

File: org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/CachingCompiler.scala [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+   *  Other thing is the update of the `AnalysisStore` implemented in `AggressiveCompile.compile1`
+   *  method which is not implemented in `IC.compile`.
+   */
+  def compile(in: SbtInputs, comps: Compilers[AnalyzingCompiler]): Analysis = {
+    val options = in.options; import options.{ options => scalacOptions, _ }
+    val aMap = (f: File) => SbtUtils.m2o(in.analysisMap(f))
+    val defClass = (f: File) => { val dc = Locator(f); (name: String) => dc.apply(name) }
+    val (previousAnalysis, previousSetup) = Cache.previousResult(cacheFile, in.incOptions.nameHashing)
+    import comps._
+    cacheAndReturnLastAnalysis(IC.incrementalCompile(scalac, javac, options.sources, classpath, output, in.cache,
```
Comment (Kwestor) [inline org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/CachingCompiler.scala]: Yay, 17 args in one method...

**Comment by wpopielarski** [comment]
About sbt part. I try to make the PR to sbt repo with this content. If it was be rejected from some more less formal reason I would go back to this solution.
Stay tune

**Comment by wpopielarski** [comment]
retest this please

**Comment by wpopielarski** [comment]
retest this please

**Comment by sschaef** [comment]
LGTM

**Pull request merged**